R/mail.R

```diff
@@ -6,110 +6,249 @@
 
 ############## Downloader ##############
 
-#' Download all pipermail files in an archive
-#' @param url An url pointing to a pipermail archive
-#' @return Returns `destination`, a vector of the downloaded files in the current working directory
+#' Download all pipermail files in an archive as mbox files
+#' @param mailing_list The name of the mailing list being downloaded
+#' @param start_year_month The year and month of the first file to be downloaded
+#' @param end_year_month The year and month of the last file to be downloaded
+#' @param save_folder_path The folder path in which all the downloaded pipermail files will be stored
+#' @return Returns `downloaded_files`, a vector of the downloaded files in the current working directory
 #' @export
-download_pipermail <- function(url) {
+download_pipermail <- function(mailing_list, start_year_month, end_year_month, save_folder_path) {
 
-  #Get page
-  pagedata <- httr::GET(url)
-
-  #Parse html file into object
-  tbls_xml <- XML::htmlParse(pagedata)
-
-  #Extract href tablenodes from html table
-  tableNodes <- XML::getNodeSet(tbls_xml, "//td/a[@href]")
+  # Create directory if it does not exist
+  if (!dir.exists(save_folder_path)) {
+    dir.create(save_folder_path, recursive = TRUE)
+  }
```

Comment:
Does this assume only the ending folder is not created? If so, I suggest you remove this code. One of the functions I will ask the config.R folks is to create a function that defines all the folder structure in Kaiaulu for the user to avoid this. So rather than us adding logic in every function for the ending folder in a path creation, the convention is enforced in a single place, the config.R, where all paths are also specified in the config.